File: app/src/main/java/com/example/cinemates/local/dao/PersonDao.java
package com.example.cinemates.local.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.cinemates.model.Cast;
import com.example.cinemates.model.Person;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;

/**
 * @author Antonio Di Nuzzo
 * Created 26/07/2022 at 07:36
 */
@Dao
public interface PersonDao {
    @Query("SELECT * FROM person WHERE favorite = 1")
    Observable<List<Person>> getAllFavorite();


    @Query("SELECT * FROM person WHERE id = :id ")
    Cast retrievePerson(Integer id);


    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Person person);


    @Delete
    void delete(Person person);
}

File: core/database/src/main/java/com/indisparte/database/dao/PersonDao.kt
package com.indisparte.database.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.indisparte.database.entity.FavoritePersonEntity

/**
 *@author Antonio Di Nuzzo
 */
@Dao
interface PersonDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertFavoritePerson(personEntity: FavoritePersonEntity): Long

    @Delete
    fun removeFavoritePerson(personEntity: FavoritePersonEntity): Int

    @Query("SELECT * FROM person ")
    fun getAllFavoritePerson(): List<FavoritePersonEntity>

    @Query("SELECT EXISTS (SELECT * FROM person WHERE id = :personId)")
    fun isPersonFavorite(personId: Int): Boolean

}